api/src/controllers/login.js

```diff
@@ -274,7 +265,7 @@ const loginByToken = (req, res) => {
       }
 
       return tokenLogin.resetPassword(userId).then(({ user, password }) => {
-        req.body = { user, password };
+        req.body = { user, password, locale: req.body.locale };
```

Comment:
This may have been a latent bug.  Several of the login tests started failing after I made the above changes and the ones involving token login (e.g. `should login the user when token is valid`) were not fixed even after I added the locale to the request body in the tests.  After some debugging I determined that the locale was getting dropped from the request body here at this line before `setCookies` is called.

Reply:
I think this is a good solution, however the body that this endpoint receives is always empty: 
https://github.com/medic/cht-core/blob/3222eb05cb8f8248485b7168505f8cfc6e4c1c31/api/src/public/login/script.js#L55

We'd need to add a payload with the selected locale there: 
```
const payload = JSON.stringify({ locale: selectedLocale });
request('POST', url, payload, xmlhttp => {
```